TripPlaner/src/main/java/com/singapore/TripPlaner/Controller/PlaceController.java

```diff
@@ -29,8 +34,24 @@ public String getPlacesByType(@RequestParam(required = true) String type, Model
         return "places";
     }
     @GetMapping ("/place/{id}")
-    public String placeDetails(@PathVariable Long id, Model model){
+    public String placeDetails(@PathVariable Long id,
+                               Model model){
         model.addAttribute("place", placeService.findById(id));
+        List opinionsId = opinionService.randomOpinions(1,id);
+        Opinion opinion = (Opinion) opinionService.findById(Double.valueOf((Double) opinionsId.get(0)).longValue());
+        model.addAttribute("opinionDetail", opinion);
+        model.addAttribute("opinion", new Opinion());
         return "placeDetails";
     }
+    @GetMapping("/places/topRate")
+    public String getTopRatePlaces(Model model) {
+        model.addAttribute("places", placeService.getTopRatedPlaces());
+        return "places";
+    }
+    @GetMapping("/places/mostPopular")
+    public String getMostPopularPlaces(Model model) {
+        model.addAttribute("places", placeService.getMostPopularPlaces());
+        return "places";
+    }
```

Comment:
to jest zbędne. Tego Twój task nie zawierał. Można z tego zrobić nowy taks w jira